src/influx_udp.erl

```diff
@@ -8,71 +6,119 @@
 
 %% ===================================================================
 %% Application callbacks
 %% ===================================================================
 
 start() ->
-  start([],[]).
+  application:start(influx_udp).
 
-start(_StartType, _StartArgs) ->
-  influx_udp_sup:start_link().
-
-stop(_State) ->
-    ok.
+stop() ->
+  application:stop(influx_udp).
 
 %% @doc
-%% Write binary data to influx.
-%% Note: Assumed that data represents Influx-valid JSON.
+%% Start new pool with Name and Options (as map)
+%% Options:
+%% - host - InfluxDB hostname
+%% - port - InflxuDB UDP port
+%% - pool_size - Pool size (numer of workers)
+%% - max_overflow - Pool max overflow size
 %% @end
+-spec start_pool(Name::atom(), Options::map()) -> ok.
+start_pool(Name, Options) ->
+  gen_server:call(?SERVER, {start_pool, Name, Options}).
 
+%% @doc
+%% Write binary data to influx using default pool.
+%% Note: Assumed that data represents InfluxDB-valid input.
+%% @end
 -spec write(Bin::binary()) -> ok.
-
 write(Bin) ->
-  gen_server:call(?SERVER, {send_to_pool, {write, Bin}}).
+  gen_server:call(?SERVER, {send_to_pool, default, {write, Bin}}).
 
 %% @doc
-%% Write data points to Series.
+%% Write binary data to influx using named pool.
 %% @end
+-spec write(Pool::atom(), Bin::binary()) -> ok
+    ; (
+        Series::atom()|string()|binary(),
+        Points::list(influx_data_point())|influx_data_point()
+      ) -> ok.
+write(Pool, Bin) when is_binary(Bin) ->
+  gen_server:call(?SERVER, {send_to_pool, Pool, {write, Bin}});
 
--spec write(Series::atom()|string()|binary(), Points::list(influx_data_point())|influx_data_point()) -> ok.
+%% @doc
+%% Write data points to Series using default pool.
+%% @end
 write(Series, Points) ->
-  gen_server:call(?SERVER, {send_to_pool, {write, Series, Points}}).
+  gen_server:call(?SERVER, {send_to_pool, default, {write, Series, Points}}).
+
+%% @doc
+%% Write data points to Series using named pool.
+%% @end
+-spec write(Pool::atom(), Series::atom()|string()|binary(), Points::list(influx_data_point())|influx_data_point()) -> ok.
+write(Pool, Series, Points) ->
+  gen_server:call(?SERVER, {send_to_pool, Pool, {write, Series, Points}}).
+
+init_server() ->
+  erlang:register(?SERVER, self()),
+  proc_lib:init_ack({ok, self()}),
+
+  %% Start default pool
+  Defaults = #{
+    port => ?Config(influx_port, 4444),
+    host => ?Config(influx_host, '127.0.0.1'),
+    pool_size => ?Config(pool_size, 3),
+    max_overflow => ?Config(max_overflow, 1)
+  },
+
+  UseDefault = ?Config(default_pool, true),
+
+  if UseDefault
+    ->  start_pool_(default, Defaults);
+    true -> ok
+  end,
+
+  gen_server:enter_loop(?SERVER, [], #state{defaults = Defaults}).
 
 %% ------------------------------------------------------------------
 %% gen_server Function Definitions
 %% ------------------------------------------------------------------
+handle_call({start_pool, Name, Options}, _, #state{defaults = Defaults} = State) ->
```

Comment:
Line 117 is too long: handle_call({start_pool, Name, Options}, _, #state{defaults = Defaults} = State) ->.